### PR TITLE
Run UI E2E tests after dev deploy

### PR DESCRIPTION
**Pull request by pezholio**
This ensures both frontends are still working as expected after any changes have been made to the API.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -196,6 +177,68 @@ jobs:
...
+          destination: playwright-report
+      - store_artifacts:
+          path: test-results
+          destination: test-results
+      - slack/notify:
+          event: fail
+          channel: << pipeline.parameters.alerts-slack-channel >>
+          template: basic_fail_1
+  cas3_e2e_tests:
+    parallelism: 4
```
Comment (tahb) [inline .circleci/config.yml:208]: Noticed CAS1 doesn't have this and it sounds useful to keep deploys faster. What's the max? Guessing it's the number of cores on whatever sized resource we pay for and so can't be increased?

File: .circleci/config.yml [hunk 2]
```diff
@@ -196,6 +177,68 @@ jobs:
...
+          name: Checkout e2e repo
+          command: |
+            git clone https://github.com/ministryofjustice/hmpps-temporary-accommodation-ui.git .
+      - run:
+          name: install
+          command: apt-get install xz-utils
+      - run:
+          name: Update npm
+          command: 'npm install -g npm@latest'
+      - node/install-packages
```
Comment (tahb) [inline .circleci/config.yml:227]: Is this our script or something that we get out of the cypress/browsers:node16.14.2-slim-chrome100-ff99-edge image, what packages does it install that's not covered by npm install? Excuse the silly question.

File: .circleci/config.yml [hunk 3]
```diff
@@ -196,6 +177,68 @@ jobs:
...
+      - run:
+          name: E2E Check - << parameters.environment >>
+          command: |
+            TESTS=$(circleci tests glob "e2e/tests/*.feature" | circleci tests split --split-by=timings | paste -sd ',')
+            npm run test:e2e:ci --\
```
Comment (tahb) [inline .circleci/config.yml:232]: I don't expect this to change much in the frontend but remembering to come and update this seems like a very easy thing to miss. Perhaps we could leave a link to this file in [our frontend e2e.env files](https://github.com/ministryofjustice/hmpps-temporary-accommodation-ui/blob/main/e2e.local.env) as a pointer?

(I assume it's impossible to load environment variables from another circleci context and that even if we could that might have unintended consequences anyway).

File: .circleci/config.yml [hunk 4]
```diff
@@ -242,6 +285,8 @@ workflows:
       - request-preprod-approval:
           type: approval
           requires:
+            - cas1_e2e_tests
```
Comment (tahb) [inline .circleci/config.yml:288]: I wonder if we should release this first without these requires so if there's a problem we don't block the pipeline. Then once it's green we follow-up with a small change to add them back in?

File: .circleci/config.yml (hunk 1 above)
Comment (pezholio) [reply .circleci/config.yml:208]: Yeah, I haven't bothered for CAS1, as we have fewer e2e tests and they're only split across two files. We could try increasing it, but it'll be diminishing returns.

File: .circleci/config.yml (hunk 2 above)
Comment (pezholio) [reply .circleci/config.yml:227]: It's a CircleCI orb thing I think, it handles caching and the like too, as well as running `npm install`

File: .circleci/config.yml (hunk 4 above)
Comment (pezholio) [reply .circleci/config.yml:288]: Good plan. In a seperate push I commented out the requires and it seems to work, but that sounds sensible too

File: .circleci/config.yml (hunk 3 above)
Comment (pezholio) [reply .circleci/config.yml:236]: I've added all the shared secrets to a [Context](https://app.circleci.com/settings/organization/github/ministryofjustice/contexts/4a855c71-1cc7-498a-b7bf-4e7d1cfbe1b1?return-to=%2F) here, so we can share them between projects.

**Review by tahb — APPROVED** [review]
👍

**Pull request merged**